homeassistant/components/iammeter/sensor.py

```diff
@@ -51,70 +114,359 @@ async def async_setup_platform(
     config_port = config[CONF_PORT]
     config_name = config[CONF_NAME]
     try:
-        async with asyncio.timeout(PLATFORM_TIMEOUT):
-            api = await real_time_api(config_host, config_port)
-    except (IamMeterError, asyncio.TimeoutError) as err:
+        api = await hass.async_add_executor_job(
+            IamMeter, config_host, config_port, config_name
+        )
+    except asyncio.TimeoutError as err:
         _LOGGER.error("Device is not ready")
         raise PlatformNotReady from err
 
     async def async_update_data():
         try:
-            async with asyncio.timeout(PLATFORM_TIMEOUT):
-                return await api.get_data()
-        except (IamMeterError, asyncio.TimeoutError) as err:
+            async with timeout(PLATFORM_TIMEOUT):
+                return await hass.async_add_executor_job(api.client.get_data)
+        except asyncio.TimeoutError as err:
             raise UpdateFailed from err
 
     coordinator = DataUpdateCoordinator(
         hass,
         _LOGGER,
-        name=DEFAULT_DEVICE_NAME,
+        name=config_name,
         update_method=async_update_data,
         update_interval=SCAN_INTERVAL,
         request_refresh_debouncer=debounce.Debouncer(
             hass, _LOGGER, cooldown=0.3, immediate=True
         ),
     )
     await coordinator.async_refresh()
-    entities = []
-    for sensor_name, (row, idx, unit) in api.iammeter.sensor_map().items():
-        serial_number = api.iammeter.serial_number
-        uid = f"{serial_number}-{row}-{idx}"
-        entities.append(IamMeter(coordinator, uid, sensor_name, unit, config_name))
-    async_add_entities(entities)
+    model = coordinator.data["Model"]
+    serial_number = coordinator.data["sn"]
+    _migrate_to_new_unique_id(hass, model, serial_number)
+    if model == DEVICE_3080:
+        async_add_entities(
+            IammeterSensor(coordinator, description)
+            for description in SENSOR_TYPES_3080
+        )
+    else:  # DEVICE_3080T:
+        async_add_entities(
+            IammeterSensor(coordinator, description)
+            for description in SENSOR_TYPES_3080T
+        )
+
 
+class IammeterSensor(update_coordinator.CoordinatorEntity, SensorEntity):
+    """Representation of a Sensor."""
 
-class IamMeter(CoordinatorEntity, SensorEntity):
-    """Class for a sensor."""
+    entity_description: IammeterSensorEntityDescription
+    _attr_has_entity_name = True
+    _attr_name = None
```

Comment:
`_attr_name` should be removed, I think. We're setting an entity description on the entity and the description either provides a translation key or a device class that defaults to a translation. By setting `_attr_name` it will override the name translations.